### PR TITLE
Added Pretty Url for Users

### LINKED ISSUE #2055: Pretty URL for Users
**Is your feature request related to a problem? Please describe.**
Having pretty URLs is a must for a website. I personally, don't like seeing random numbers appear in the address bar. It would be better if we have pretty URLs for a user.

**Describe the solution you'd like**
`/users/123434234` can be replaced by `/users/:user-name` by adding a slug to users' schema or by using [friendly_id](https://github.com/norman/friendly_id) which has already been used a few times in this project.

**Additional context**

 _Before_
> before

_After_
> 
Screenshot from 2021-01-03 01-00-30

**Are you working on this? (Yes/No)**
Yes

### PR DESCRIPTION
**Pull request by Prerna-0202**
Fixes #2055 

#### Describe the changes you have made in this PR -
I have added the feature "Pretty Url for Users" as per issue #2055 

### Screenshots of the changes (If any) -
**Before:-**
Screenshot (340)

**After:-**
Screenshot (339)

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

File: config/initializers/friendly_id.rb [hunk 1]
```diff
@@ -44,14 +44,14 @@
   # performance because it will avoid Rails-internal code that makes runtime
   # calls to `Module.extend`.
   #
-  # config.use :finders
+  config.use :finders
```
Comment (tachyons) [inline config/initializers/friendly_id.rb:47]: Don't use finders module, it does have side effects

Comment (Prerna-0202) [reply config/initializers/friendly_id.rb]: Okay, I removed it and checked that it is working fine...

**Comment by tachyons** [comment]
The main reason we kept holding this issue was that urls can be too long as names are not unique, could you check how the URLs looks like there are two people with same name and same project name ?

**Comment by Prerna-0202** [comment]
> The main reason we kept holding this issue was that urls can be too long as names are not unique, could you check how the URLs looks like there are two people with same name and same project name ?

yaa right, URLs are too long for two same users. 
Screenshot (347)
Screenshot (348)
Screenshot (351)
Screenshot (354)

**Comment by vedant-jain03** [comment]
> > The main reason we kept holding this issue was that urls can be too long as names are not unique, could you check how the URLs looks like there are two people with same name and same project name ?
> 
> So, we have to make URLs short for the same name users?

What if we can do this:
if the user has the same name then URL could be `${url}/name-${user.created_at}`
created_at field will be different and it will not make the URL larger as it does in above-mentioned sample.
WDYT? @tachyons

**Comment by tachyons** [comment]
@vedant-jain03 That will make URLs unnecessarily complex, I am closing this for now. We can revisit them once we have unique username kind of mechanism

**Pull request closed without merging**